packages/fxa-react/lib/utils.tsx

```diff
@@ -12,3 +14,161 @@ export type FtlMsgProps = {
+    public readonly strict: boolean
+  ) {}
+
+  /**
+   * A wrapper around l10n.getString that provides more safety. When strict is true, using this function ensures:
+   *  - The fallback text is in alignment with the default 'en' text
+   *  - There are no missing parameters
+   *  - There are no straight quotes
+   *
+   * Note, this requires that the default 'en' bundle was loaded and is a available.
+   *
+   * @param id - The id of the message to fetch
+   * @param fallbackText - The fallback text for the message
+   * @param args - Any arguments for parameterization of message.
+   * @returns The formatted language translation string.
+   */
+  public getMsg(
+    id: string,
+    fallbackText: string,
+    args?: Record<string, FluentVariable>
+  ) {
+    // If strict, perform checks to validate that state of request is valid.
+    if (this.strict) {
+      if (!id) {
+        throw new Error('An l10n id must be provided.');
+      }
+
+      // Find an 'en' bundle with the message id. The 'en' bundle should be the default and always exist.
+      const enBundles = [...this.l10n.bundles].filter((x) =>
+        x?.locales?.includes('en')
+      );
+
+      if (enBundles.length === 0) {
+        throw new Error(
+          `No 'en' bundles loaded. The 'en' bundle is the default and should always be loaded.`
+        );
+      }
+
+      // Make sure en bundle has the target message
+      const enBundle = enBundles.find((x) => x.hasMessage(id));
+      const enMessage = enBundle?.getMessage(id);
+      if (!enBundle || !enMessage) {
+        throw new Error(
+          `Could not locate message in en bundle. Message id: ${id}`
+        );
+      }
+
+      // Make sure the fallback text and message render cleanly.
+      checkMessage(enBundle, enMessage, fallbackText, args);
+    }
+
+    // If the above checks pass, we can safely return the language string.
+    return this.l10n.getString(id, args, fallbackText);
+  }
+}
+
+/**
+ * Validates a message. Checks that:
+ *  - fallback text matches ftl text
+ *  - there are not straight quotes
+ *  - there are no missing ftlArgs
+ * @param bundle - The default bundle, typically a bundle with 'en' locale.
+ * @param pattern - The pattern to check. This is the fluent messages's 'value'
+ * @param fallbackText - The fallback text
+ * @param ftlArgs - Arguments for updating parameterized messages
+ */
+export function checkMessage(
+  bundle: FluentBundle,
+  message: Message,
+  fallbackText: string | null,
+  ftlArgs?: Record<string, FluentVariable>
+) {
+  if (!fallbackText) {
+    throw new Error('Fallback text must be provided.');
+  }
+
+  // nested attributes can happen when we define something like:
+  // `profile-picture =
+  //   .header = Picture`
+  const nestedAttrValues = Object.values(message?.attributes || {});
+
+  if (
+    message === undefined ||
+    (message.value === null && nestedAttrValues.length === 0)
+  ) {
+    throw Error(`Invalid message. Message has no value.`);
```

Comment:
Re: consistency w/ L98 above:
```suggestion
    throw new Error(`Invalid message. Message has no value.`);
```

Curious if https://eslint.org/docs/latest/rules/no-throw-literal would catch something like this. Since `throw Error()` isn't exactly a literal, but I think we generally use `throw new Error()` (or something like `throw OauthError.incorrectSecret()`).

Reply:
Getting off track a bit, but here's the current state of the #main branch.

```sh
git grep -In 'throw Error[(]' packages | cat

# packages/fxa-auth-server/lib/routes/oauth/token.js:197:        throw Error('unreachable');
# packages/fxa-event-broker/src/pubsub-proxy/pubsub-proxy.controller.ts:200:        throw Error(`Invalid event: ${message.event}`);
# packages/fxa-payments-server/src/lib/test-utils.tsx:700:    throw Error(`unable to locate fluent message with id: ${msgId}`);
# packages/fxa-profile-server/lib/events.js:22:      throw Error('Unable get uid from message event ' + message.event);
# packages/fxa-react/lib/test-utils/index.ts:102:    throw Error(
# packages/fxa-react/lib/test-utils/index.ts:108:    throw Error(
# packages/fxa-react/lib/test-utils/index.ts:114:    throw Error(
# packages/fxa-react/lib/test-utils/index.ts:164:    throw Error(`Could not retrieve Fluent message tied to ID: ${ftlId}`);
```

Reply:
Okay, should be consistent within fxa-react now.